consumer.go

```diff
@@ -104,6 +105,23 @@ func (c *Consumer) Scan(ctx context.Context, fn ScanFunc) error {
 		shardc = make(chan *kinesis.Shard, 1)
 	)
 
+	// Discover the retention period on the stream, which is used later in ScanShard()
+	// for failure mode handling.
+	summary, err := c.client.DescribeStreamSummaryWithContext(
+		aws.Context(ctx),
+		&kinesis.DescribeStreamSummaryInput{
+			StreamName: aws.String(c.streamName),
+		},
+	)
+	if err != nil {
+		if err.(awserr.Error).Code() != "AccessDeniedException"  {
```

Comment:
There is no constant for this that I could find.  `kinesis.ErrCodeAccessDeniedException` isn't a thing, and while this constant is defined in other packages, I didn't think it made sense to import one of them just for that.